File: app/src/main/java/com/sample/dhruvmj/waveapp/util/NetworkUtil.java
package com.sample.dhruvmj.waveapp.util;

import android.content.Context;
import android.net.ConnectivityManager;

/**
 * @author Dhruv
 */

public class NetworkUtil {
    /**
     * This method will check whether the internet is connected
     *
     * @param context This is the context from which
     *                this method is invoked
     * @return a boolean flag
     */
    public static boolean isNetworkConnected(Context context) {
        ConnectivityManager connectivityManager = ((ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
        return connectivityManager != null && connectivityManager.getActiveNetworkInfo() != null && connectivityManager.getActiveNetworkInfo().isConnected();
    }
}

File: app/src/main/java/com/karlptrck/wavechallenge/util/NetworkUtil.kt
package com.karlptrck.wavechallenge.util

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

class NetworkUtil {

    companion object {

        fun isConnectionAvailable(context: Context): Boolean {
            val connectivity = context.getSystemService(
                Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            if (connectivity != null) {
                val info = connectivity.allNetworkInfo
                if (info != null)
                    for (i in info)
                        if (i.state == NetworkInfo.State.CONNECTED) {
                            return true
                        }
            }
            return false
        }
    }
}